### DISCUSSION TITLE
How to use the name of the input in VRL?

### DISCUSSION THREAD

**Discussion by smitsr72** [vectordotdev/vector#11934] [Q&A] [answered]
How can I write and if statement using the name of the input to make a decision?

**Comment by StephenWakely** [comment, marked as answer]
So, supposing you had a field called `name` in your input, you could write something like the following:

```coffee
if .name == "flub" {
  # Do the thing when name is flub.
} else {
  # Do the thing when name is not flub.
}
```

You can find more details [here](https://vector.dev/docs/reference/vrl/expressions/#if).

**Reply by smitsr72** [reply]
Sorry for the confusion. I want to make a decision based on the name of the input source in a remap vrl.

With the remap transformation the inputs are listed like this:
inputs = ["A", "B", "C"]

Then in the vrl code I want to make a decision based on source A, B or C.

``
type = "remap"
inputs = ["A", "B", "C"]
source = '''
  if **metadata.input** == "A" 
  {
    .bla = "A" 
  }
  else if **metadata.input** == "B" 
  {
    .bla = "B" 
  }
  else 
  {
    .bla = "N/A"
  }
  '''
``

The **metadata.input** does not exist, I just made that up, to give the idea what I would like to do.

Alternatively, How can I inject the name of the source "A" into the output of the source? 
```
[sources.A]
type = "file"
line_delimiter = "\r\n"
include = [ "C:/logs/*.log" ]
read_from = "beginning"
```

**Reply by StephenWakely** [reply]
Ah, I see what you mean.  Your best bet would be to have a separate remap transform for each input that adds a field to tag the event with the source.

```
source = '''
  .input = "A"
'''
```

Then the final remap can work with this field.

**Reply by smitsr72** [reply]
I also thought of that workaround. I have many input sources, which means also many remaps just for the purpose of tagging the source in the message output. I will apply this strategy and see how it works out.

**Reply by StephenWakely** [reply]
I have raised a feature request [here](https://github.com/vectordotdev/vector/issues/11935) to see if we can come up with a better solution for this situation. Feel free to add any further context to that.
